src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -208,10 +176,19 @@ export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, sh
     )(async () => {
       try {
         const workspace = await Ajax(signal).Workspaces.workspace(namespace, name).details()
-        const res = await checkBucketAccess(signal, namespace, name)
-        workspaceStore.set({ hasBucketAccess: res, ...workspace })
+        workspaceStore.set(workspace)
 
         const { accessLevel, workspace: { createdBy, createdDate } } = workspace
+
+        // Request a service account token. If this is the first time, it could take some time before everything is in sync.
+        // Doing this now, even though we don't explicitly need it now, increases the likelihood that it will be ready when it is needed.
+        if (Utils.canWrite(accessLevel)) {
+          saToken(namespace)
+        } else {
+          // Request a "default" service account token to prevent proliferation of service accounts in public workspaces
+          saToken()
```

Comment:
Is this something we need to do, or does this effectively get initiated somewhere else in the process of setting up a user?

Also of note, we don't use this default service account anywhere when making bucket calls from the UI. Maybe we should, but that would be a much larger change.

Reply:
No, this is the thing that will on-demand create a pet service account if necessary. If that doesn't happen until someone tries to run an analysis, there's a chance that the analysis will fail while things are syncing. The diagnostic bucket read effectively did this so if we're not going to call that, we should at least do this to get the pet ready to go.

Right. The place we would use this is if we fully implement the diagnostic bucket read, though that then has implications for requester pays buckets. Rather than do that, I can remove the "default" service account code and only prefetch the token if the user has write access. Sound good?